rundeckapp/grails-app/services/rundeck/services/ExecutionUtilService.groovy

```diff
@@ -336,12 +339,10 @@ class ExecutionUtilService {
      * @param path path to store the file on filesystem. If null a temporary file will be created and deleted.
      * @return file containing execution.xml
      */
-    File getExecutionXmlFileForExecution(Execution execution, String path = null) {
-        File executionXmlfile
-        if(path){
-            executionXmlfile  = new File(path, "${execution.id}.execution.xml")
-        }else{
+    File getExecutionXmlFileForExecution(Execution execution, File executionXmlfile = null) {
```

Comment:
getExecutionXmlFileForExecution is not used anywhere except here https://github.com/rundeck/rundeck/pull/5966/files#diff-d5afe3953d1f462e7d2662435b77507eR81

so i would remove the temp file creation altogether

Reply:
done